spec/requests/users_controller_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::UsersController, type: :request do
+  let(:valid_user_params) do
+    {
+      user: {
+        name: Faker::Name.name,
+        email: Faker::Internet.email,
+        password: 'Password123+',
+        password_confirmation: 'Password123+'
+      }
+    }
+  end
+
+  describe 'Signup' do
+    let(:headers) { { 'ACCEPT' => 'application/json' } }
+
+    context 'valid user params' do
+      it 'creates a user account for valid params' do
+        expect { post api_v1_users_path, params: valid_user_params, headers: }.to change(User, :count).from(0).to(1)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:created)
+        expect(JSON.parse(response.body)['errors']).to be_empty
+      end
+    end
+
+    context 'invalid user params' do
+      it 'fails for invalid values' do
+        invalid_user_params = {
+          user: { name: '', email: 'invalid', password: 'something', password_confirmation: 'something' }
+        }
+        user = User.new invalid_user_params[:user]
+        user.provider = 'greenlight'
+        expect(user).not_to be_valid
+        expect { post api_v1_users_path, params: invalid_user_params, headers: }.not_to change(User, :count)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors']).to eq user.errors.to_a
+      end
+
+      it 'fails for password confirmation mismatches' do
+        invalid_user_params = valid_user_params
+        invalid_user_params[:user][:password_confirmation] = 'something_else'
+        user = User.new invalid_user_params[:user]
+        user.provider = 'greenlight'
+        expect(user).not_to be_valid
+        expect { post api_v1_users_path, params: invalid_user_params, headers: }.not_to change(User, :count)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors']).to eq user.errors.to_a
```

Comment:
Would it better to remove it entirely since we agreed that each check is done by the model itself?
I think its better to keep only two examples in the integration (controller) spec where one tests the success of the full signup operation with valid data and the other tests its fail for invalid data without specifying what invalid means here.
And add a new example in the model spec where we test the password confirmation process?